### PR TITLE
Use GitVersion to compare versions using semver

### LINKED ISSUE #25: kube version has a plus in the minor version string
The response from `kubeVersion.Minor` [here](https://github.com/hickeyma/helm-mapkubeapis/blob/master/pkg/common/common.go#L117) will be e.g `16+` rather than `16`. This means that `strconv.ParseFloat` fails, as `1.16+` cannot be properly parsed.

I'm creating an issue because I'm not certain of the best way to tackle this – do we just want to drop the `+`? Should we rely on `GitVersion` instead?

The advantage of `GitVersion` would be that we can do a proper semver compare, as versions like `v1.16.6-beta.0` or `v1.16.8-gke.15` are both valid SemVer versions, and can be properly compared against e.g `"1.9"`. This would also make the `Map.yaml` a bit more readable as we could do `"1.9"` rather than `"1.09"` which looks a bit weird (after all, it _is_ version 1.9).

Happy to PR this once there's some decision on how we want to do the version comparison :)

### PR DESCRIPTION
**Pull request by flowjo-shaunr**
Fixes #25

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,6 +12,7 @@ require (
 	github.com/rubenv/sql-migrate v0.0.0-20200402132117-435005d389bc // indirect
 	github.com/spf13/cobra v0.0.5
 	github.com/spf13/pflag v1.0.5
+	golang.org/x/mod v0.1.1-0.20191105210325-c90efee705ee
```
Comment (hickeyma) [inline go.mod:15]: Should this be v0.3.0?

File: pkg/common/common.go [hunk 2]
```diff
@@ -65,8 +64,7 @@ func ReplaceManifestUnSupportedAPIs(origManifest, mapFile string, kubeConfig Kub
 	if err != nil {
 		return "", err
 	}
-	kubeVersion, err := strconv.ParseFloat(kubeVersionStr, 32)
-	if err != nil {
+	if !semver.IsValid(kubeVersionStr) {
 		errors.Wrap(err, "Failed to get Kubernetes server version")
```
Comment (hickeyma) [inline pkg/common/common.go:68]: errors.Wrap(err, "Failed to get Kubernetes server version") --> errors.Errorf("Failed to get Kubernetes server version")

File: pkg/common/common.go [hunk 3]
```diff
@@ -80,8 +78,7 @@ func ReplaceManifestUnSupportedAPIs(origManifest, mapFile string, kubeConfig Kub
 		} else {
 			apiVersionStr = mapping.RemovedInVersion
 		}
-		apiVersion, err := strconv.ParseFloat(apiVersionStr, 32)
-		if err != nil {
+		if !semver.IsValid(apiVersionStr) {
 			errors.Wrapf(err, "Failed to get the deprecated or removed Kubernetes version for API: %s", deprecatedAPI)
```
Comment (hickeyma) [inline pkg/common/common.go:82]: errors.Wrapf(err, "Failed to get the deprecated or removed Kubernetes version for API: %s", deprecatedAPI) --> errors.Errorf("Failed to get the deprecated or removed Kubernetes version for API: %s", deprecatedAPI)

Comment (eloo) [reply pkg/common/common.go:82]: also the output of errors.Errorf or Wrap should be handled.. 
right now it will be ignored AFAIK

Comment (flowjo-shaunr) [reply pkg/common/common.go:82]: What behavior do you think would be best for the error in the for loop? Return immediately if an error is found from the map file, or wrap all and return the error following the loop?

Comment (flowjo-shaunr) [reply pkg/common/common.go:82]: It looks like currently it wraps all errors but then the error isn't returned

Comment (eloo) [reply pkg/common/common.go:82]: IMHO i would recomment to return it as this behaviour is also used a few lines above too.
so 
```
return "", errors.Errorf(...)
```

Comment (flowjo-shaunr) [reply pkg/common/common.go]: Okay, I've changed error behavior to return immediately and improved the output formatting of the API details a bit

**Pull request merged**